shell/platform/darwin/macos/framework/Headers/FlutterEngine.h

```diff
@@ -89,9 +91,40 @@ FLUTTER_DARWIN_EXPORT
  *
  * Setting this field from non-nil to a different non-nil FlutterViewController
  * is prohibited and will throw an assertion error.
+ *
+ * This method is deprecated. Querying view controllers should use
+ * viewControllerForId: instead. Assigning or replacing view controllers do not
+ * have a replacement. Consider addViewController: and removeViewController:.
  */
 @property(nonatomic, nullable, weak) FlutterViewController* viewController;
 
+/**
+ * Attach a view controller to the engine and associate it with a newly
+ * generated ID.
+ *
+ * The engine holds a weak reference to each attached view controller.
+ *
+ * The first added view controller (either with this method or the
+ * viewController property) will always have ID kFlutterDefaultViewId.
```

Comment:
I would consider this an implementation detail and not document this. People shouldn't assume that a view will get a certain ID. They should always read the ID from the view controller.

Reply:
I think this detail is actually pretty important because it means that the first `addViewController` is equivalent to `setViewController`, which matters if people are migrating from `setViewController` to `addViewController`. To put it in another way, if you do a `addViewController` first for your engine, then `setViewController` always operates that FVC.